src/s3.ts

```diff
@@ -1,24 +1,65 @@
-import { S3Client } from "@aws-sdk/client-s3";
 import { fromEnv } from "@aws-sdk/credential-providers";
+import { Disk } from "flydrive";
+import { FSDriver } from "flydrive/drivers/fs";
+import { S3Driver } from "flydrive/drivers/s3";
+import type { DriverContract } from "flydrive/types";
 
 // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
-const region = process.env["S3_REGION"];
+export const assetPath = process.env["FS_ASSET_PATH"];
 
-export const s3 = new S3Client({
-  credentials: fromEnv(),
-  region: region == null || region === "" ? "auto" : region,
-  // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
-  endpoint: process.env["S3_ENDPOINT_URL"],
-  // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
-  forcePathStyle: process.env["S3_FORCE_PATH_STYLE"] === "true",
-});
+// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+const region = process.env["S3_REGION"];
 
 // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
 const bucket = process.env["S3_BUCKET"];
-if (bucket == null) throw new Error("S3_BUCKET is required");
-export const S3_BUCKET = bucket;
 
 // biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
-const urlBase = process.env["S3_URL_BASE"];
-if (urlBase == null) throw new Error("S3_URL_BASE is required");
-export const S3_URL_BASE = urlBase;
+export const assetUrlBase = process.env["ASSET_URL_BASE"];
+if (assetUrlBase == null) throw new Error("ASSET_URL_BASE is required");
+export const S3_URL_BASE = assetUrlBase;
+
+// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+const endpointUrl = process.env["S3_ENDPOINT_URL"];
+
+// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+const accessKeyId = process.env["AWS_ACCESS_KEY_ID"];
+
+// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+const secretAccessKey = process.env["AWS_SECRET_ACCESS_KEY"];
+
+// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
+export const driveDisk = process.env["DRIVE_DISK"];
+if (driveDisk == null) throw new Error("DRIVE_DISK is required");
```

Comment:
As there are existing users that use S3, it should default to `"s3"` for backward compatibility.  Instead, we could record a warning message instead:

```suggestion
// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
if (process.env["DRIVE_DISK"] == null) {
  getLogger(["hollo", "assets"]).warn(
    "DRIVE_DISK is not configured; defaults to 's3'.  " +
    "The DRIVE_DISK environment variable will be mandatory in the future versions."
  );
}
// biome-ignore lint/complexity/useLiteralKeys: tsc complains about this (TS4111)
export const driveDisk = process.env["DRIVE_DISK"] ?? "s3";
if (driveDisk == null) throw new Error("DRIVE_DISK is required");
```